src/Plugin/media/Source/Media.php

```diff
@@ -519,4 +535,278 @@ private function getMediaValue(MediaInterface $media, $attribute_name) {
+        $field_storage->getSetting('target_type') === 'media') {
+        $options[$field_name] = sprintf('%s (%s)', $field->getLabel(), $field_name);
+      }
+    }
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    parent::submitConfigurationForm($form, $form_state);
+    /** @var \Drupal\media\MediaTypeInterface $entity */
+    $entity = $form_state->getFormObject()->getEntity();
+
+    // If a media bundle and field is chosen to map the thumbnail to, and no
+    // entity reference field was chosen, create an entity reference field.
+    $configuration = $this->getConfiguration();
+    if (!empty($configuration['media_image_bundle']) &&
+      !empty($configuration['media_image_field']) &&
+      empty($this->configuration['media_image_entity_reference_field'])) {
+      // Create the field storage.
+      $field_storage = $this->createMediaImageEntityReferenceFieldStorage();
+      // Create the field.
+      $this->createMediaImageEntityReferenceField($entity, $field_storage);
+      $this->configuration['media_image_entity_reference_field'] = $field_storage->getName();
+    }
+  }
+
+  /**
+   * Create the field storage for storing and entity reference to a media image.
+   *
+   * @return \Drupal\field\FieldStorageConfigInterface
+   *   The created field storage for the media image entity reference.
+   */
+  protected function createMediaImageEntityReferenceFieldStorage() {
+    /** @var \Drupal\field\FieldStorageConfigInterface $field_storage */
+    $field_storage = $this->entityTypeManager
+      ->getStorage('field_storage_config')
+      ->create([
+        'entity_type' => 'media',
+        'field_name' => $this->getMediaImageEntityReferenceFieldName(),
+        'type' => 'entity_reference',
+        'settings' => [
+          'target_type' => 'media',
+        ],
+      ]);
+    $field_storage->save();
+    return $field_storage;
+  }
+
+  /**
+   * Create the field for storing an entity reference to the mpx thumbnail.
+   *
+   * @param \Drupal\media\Entity\MediaType $media_type
+   *   The mpx media type to create the field for.
+   * @param \Drupal\field\FieldStorageConfigInterface $field_storage
+   *   The field storage to use for the created field.
+   *
+   * @return \Drupal\field\FieldConfigInterface
+   *   The field config that was created.
+   */
+  protected function createMediaImageEntityReferenceField(MediaType $media_type, FieldStorageConfigInterface $field_storage) {
+    /** @var \Drupal\field\FieldConfigInterface $field */
+    $field = $this->entityTypeManager
+      ->getStorage('field_config')
+      ->create([
+        'field_storage' => $field_storage,
+        'bundle' => $media_type->id(),
+        'label' => $this->t('Media image'),
+        'required' => FALSE,
+      ]);
+    $field->save();
+    return $field;
+  }
+
+  /**
+   * Generate a name for a new entity reference field.
```

Comment:
I'm surprised core doesn't have something for this, given media module has to do this when creating a new bundle.

Reply:
I know right? The MediaSourceBase has it's own protected method for doing the same. This was mostly inspired by that method: `\Drupal\media\MediaSourceBase::getSourceFieldName`.